service/search/search.go

```diff
@@ -132,6 +132,18 @@ func (s *SearchOp) Run(ctx context.Context, adaptor proc.DataAdaptor, pool *lake
 	return driver.RunWithLakeAndStats(ctx, d, seq, zctx, adaptor, statsTicker.C, s.logger, parallelism)
 }
 
+func poolHasSpan(ctx context.Context, pool *lake.Pool) bool {
```

Comment:
I think this can just check len(pool.Layout.Keys) == 0

Reply:
That'll break the app in a different way since it currently creates pools with no key.